### PR TITLE
[API + Events] [Additions] Role API + More extensions

### PR DESCRIPTION
**Pull request by Misfiy**
Additions:
* ``CollectionExtensions::RemoveSpecified(IEnumerable<T>, Func<T, bool>)``
* ``EnumExtensions::QueryEnumValue<T>()`` -> Queries the values of an Enum.
* ``RoleExtensions::GetRandom(Team)``
* ``RoleExtensions::GetRandom(Side)``
* ``Role::AllRoles`` -> a list of all RoleTypeId.
* ``Role::ShuffledAllRoles`` -> A shuffled version of ``Role::AllRoles``
* ``Role::NextScpSpawn``
* ``Role::NextHumanSpawn``
* ``Scp079Role::Scp079ElevatorStateChanger``
* ``Scp079Role::TeslaCooldown`` &``Scp079Role::TeslaUseCost``
* ``Scp079Role::SpeakingRegenMultiplier``
* ``Scp079Role::BlackoutCost``, ``Scp079Role::BlackoutDuration``, ``Scp079Role::BlackoutCooldown``
* ``Scp079Role::ElevatorCost``
* ``LosingSignal`` & ``LostSignal`` event (for Scp079)

Fixes:
* Possibly fix ``Scp3114Role::StolenRole`` when not already disguised

Breaking changes:
* ``Scp079Role::LockDoor(Door)`` -> ``Scp079Role::LockDoor(Door, bool)`` (both exist, first is being removed)
* Rename ``Role::Random`` -> ``Role::GetRandom``

### CODE REVIEW

**Comment by Misfiy** [comment]
> Can you also add `Scp079LostSignalHandler::_ghostlightLockoutDuration`?
Scp2176LostTime exists already for this, would you prefer a rename?

**Comment by VALERA771** [comment]
No, missed it

**Comment by Misfiy** [comment]
Lemme know what else you want for other roles too :3

File: Exiled.API/Features/Roles/Scp079Role.cs [hunk 1]
```diff
@@ -389,41 +483,13 @@ public float Scp2176LostTime
...
-                DoorLockChanger._lockTime = NetworkTime.time;
-                DoorLockChanger.LockedDoor.ServerChangeLock(DoorLockReason.Regular079, true);
-                if (door.Room is not null)
-                    MarkRoom(door.Room);
-                AuxManager.CurrentAux -= DoorLockChanger.GetCostForDoor(DoorAction.Locked, DoorLockChanger.LockedDoor);
-                return true;
-            }
-
-            return false;
-        }
```
Comment (louis1706) [inline Exiled.API/Features/Roles/Scp079Role.cs:426]: Where is LockDoor ?

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Pull request merged**